--- editors/ckeditor.inc	Sat Nov 05 03:59:33 2011
+++ editors/ckeditor.inc	Tue Nov 08 09:36:01 2011
@@ -180,6 +180,14 @@
     }
   }
 
+  // Get additional styles, supported in ckeditor versions 3.2.1 and above.
+  if (!empty($config['css_classes'])) {
+    $styles = wysiwyg_ckeditor_get_styles_set($config['css_classes']);
+    if (!empty($styles)) {
+      $settings['stylesSet'] = $styles;
+    }
+  }
+
   if (isset($config['language'])) {
     $settings['language'] = $config['language'];
   }
@@ -331,3 +339,37 @@
   return $plugins;
 }
 
+/**
+ * Converts the CSS classes string to an array of js settings.
+ *
+ * The CSS classes are edited in element CSS Classes, fieldset CSS
+ * of page admin/config/content/wysiwyg/profile/{filter}/edit.
+ *
+ * Note: only checking for correct format of each line. No error
+ * logging/hinting, as that should be done on the settings form.
+ *
+ * @param string $css_classes
+ *   A list of classes to add to the style drop down.
+ * @return array
+ */
+function wysiwyg_ckeditor_get_styles_set($css_classes) {
+  $css_styles = array();
+  $lines = explode("\n", $css_classes);
+  foreach ($lines as $line) {
+    if (!empty($line)) {
+      $l = explode('=', $line);
+      if (count($l) === 2) {
+        $lc = explode('.', $l[1]);
+        if (count($lc) === 2) {
+          $style['name'] = trim($l[0]);
+          $style['element'] = trim($lc[0]);
+          $style['attributes']['class'] = trim($lc[1]);
+          if (!empty($style['name']) && !empty($style['element']) && !empty($style['attributes']['class'])) {
+            $css_styles[] = $style;
+          }
+        }
+      }
+    }
+  }
+  return $css_styles;
+}
--- editors/js/ckeditor-3.0.js	Sat Nov 05 03:59:33 2011
+++ editors/js/ckeditor-3.0.js	Tue Nov 08 09:36:22 2011
@@ -23,6 +23,10 @@
         }
       }
     }
+    // Register Font styles (versions 3.2.1 and above).
+    if (CKEDITOR.stylesSet && Drupal.settings.wysiwyg.configs.ckeditor[format].stylesSet) {
+      CKEDITOR.stylesSet.add(format, Drupal.settings.wysiwyg.configs.ckeditor[format].stylesSet);
+    }
   }
 };
 
--- wysiwyg.admin.inc	Sat Nov 05 03:59:33 2011
+++ wysiwyg.admin.inc	Tue Nov 08 09:35:44 2011
@@ -286,7 +286,11 @@
     '#type' => 'textarea',
     '#title' => t('CSS classes'),
     '#default_value' => $profile->settings['css_classes'],
-    '#description' => t('Optionally define CSS classes for the "Font style" dropdown list.<br />Enter one class on each line in the format: !format. Example: !example<br />If left blank, CSS classes are automatically imported from all loaded stylesheet(s).', array('!format' => '<code>[title]=[class]</code>', '!example' => 'My heading=header1')),
+    '#description' => t('Optionally define CSS classes for the "Font style" dropdown list.<br />Enter one class on each line in the format: !format. Example: !example<br />If left blank, CSS classes are automatically imported from all loaded stylesheet(s).',
+      $profile->editor == 'ckeditor' && version_compare($editor['installed version'], '3.2.1', '>=')
+      ? array('!format' => '<code>[title]=[element].[class]</code>', '!example' => 'My heading=h1.header1')
+      : array('!format' => '<code>[title]=[class]</code>', '!example' => 'My heading=header1')
+    ),
   );
 
   $form['submit'] = array(
