diff --git a/mediaelement.module b/mediaelement.module
index 822941d..3b7bc84 100644
--- a/mediaelement.module
+++ b/mediaelement.module
@@ -143,9 +143,17 @@ function mediaelement_field_formatter_view($entity_type, $entity, $field, $insta
     }
 
     $class = 'mediaelement-formatter-identifier-' . time() . '-' . $id++;
+    switch ($field['type']) {
+      case 'file':
+        $source = file_create_url($item['uri']);
+        break;
+      case 'link_field':
+        $source = $item['url'];
+        break;
+    }
     $element[$delta] = array(
       '#attributes' => array(
-        'src' => file_create_url($item['uri']),
+        'src' => $source,
         'class' => $class,
       ),
       '#settings' => $settings,
